### ISSUE TITLE
Cake 2 - getAffectedRows can return 0 when query is ran via Model->query()

### ISSUE THREAD

**Issue by AshleyPinner** [cakephp/cakephp#11615]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 2.10.6

### What you did
I have a shell that generates some SQL (a MySQL INSERT IGNORE INTO query) that runs via:

```PHP
$rows = 0;
$sql = $this->obtainSQL($activityType, $lastRun); #returns valid SQL
$result = $this->Model->query($sql);
if($result) {
    $rows = $this->Model->getAffectedRows();
}
```
(Actual code, Model is a different model, but `query` and `getAffectedRows` are not overridden anywhere)

### What happened
`$rows` is always 0

### What you expected to happen
`$rows` to be non-zero when the INSERT IGNORE query inserts records.

So what I suspect is happening is that when you call getAffectedRows, you end up calling [DboSource::hasResult](https://github.com/cakephp/cakephp/blob/2.x/lib/Cake/Model/Datasource/DboSource.php#L946-L949)

The issue is, that [DboSource::_execute can sometimes move the result into `$this->_results`](https://github.com/cakephp/cakephp/blob/2.x/lib/Cake/Model/Datasource/DboSource.php#L473), rather than `$this->_result` (note the extra `s`).

This means that when you call $this->Model->getAffectedRows();, it thinks that DboSource::hasResult is false, and just returns 0 (Ref [DboSource::lastAffected](https://github.com/cakephp/cakephp/blob/2.x/lib/Cake/Model/Datasource/DboSource.php#L519-L524))

There's only one `$this->_results` in all of the DboSource.php file, and there's no tests at all on `getAffectedRows`, `hasResult` and the tests for `lastAffected` don't try a query run via `Model->query()`

**Comment by markstory** [comment]
Good eye on finding this. Given that you know how to reproduce this issue would you be able to put together a pull request to fix it?

**Comment by markstory** [comment]
Fixed in 581614349334a09580b80fa7e5369ac79ce325ab
